<launch>
  <arg name="output" default="screen"/>
  <arg name="bag"/>
  <arg name="out_file_base" default="video_"/>
  <arg name="write_time_stamps" default="true"/>
  <arg name="convert_to_mp4" default="true"/>
  <arg name="start_time" default="0.0"/>
  <arg name="duration" default="1000000.0"/>
  <arg name="video_rate" default="30"/>

  <!-- nodes -->
  <node pkg="ffmpeg_image_transport_tools" type="extract_ffmpeg"
	name="extract_ffmpeg" output="$(arg output)">
    <param name="bag_file" type="string" value="$(arg bag)"/>
    <param name="out_file_base" type="string" value="$(arg out_file_base)"/>
    <!-- will stop after max_num_frames -->
    <param name="max_num_frames" type="int" value="1000000000"/>
    <param name="start_time" type="double" value="$(arg start_time)"/>
    <param name="duration" type="double" value="$(arg duration)"/>
    <param name="video_rate" type="double" value="$(arg video_rate)"/>

    <param name="write_time_stamps" value="$(arg write_time_stamps)"/>
    <param name="convert_to_mp4" value="$(arg convert_to_mp4)"/>
  </node>
</launch>
